kafka-event-listener/src/main/java/org/hypertrace/core/kafka/event/listener/KafkaLiveEventListener.java

```diff
@@ -30,20 +30,29 @@
  * for sample usage and test. Note that testing requires Thread.sleep > poll timeout in between
  */
 public class KafkaLiveEventListener<K, V> implements AutoCloseable {
+  Queue<BiConsumer<? super K, ? super V>> callbacks;
   private final Future<Void> kafkaLiveEventListenerCallableFuture;
   private final ExecutorService executorService;
   private final boolean cleanupExecutor;
 
   private KafkaLiveEventListener(
       KafkaLiveEventListenerCallable<K, V> kafkaLiveEventListenerCallable,
+      Queue<BiConsumer<? super K, ? super V>> callbacks,
       ExecutorService executorService,
       boolean cleanupExecutor) {
+    this.callbacks = callbacks;
```

Comment:
As we had discussed the other day either here or in the builder (ideally here), we should copy off the array.

The usage of a concurrent queue is an impl detail of this class only, so shouldn't rely on a concurrent impl being passed in - the builder didn't even need to change since it doesn't have its own requirement for concurrency.